### ISSUE TITLE
Holdout Workflow RenderManForBlender

### ISSUE THREAD

**Issue by SeanMorgan1962** [prman-pixar/RenderManForBlender#568]
I am new to using holdouts in RManForBlender.  Have tried all the workflow suggestions I have found and none seem to be working correctly.  I have never used separate passes and combined them so, I have to assume I am not doing something right.  Would be grateful for some assistance.  Thank you.

P.S.  This is the rgba pass and the holdout is black.  

![scene 0024](https://user-images.githubusercontent.com/37384828/37431786-6f536ca4-2793-11e8-8ac0-1c6b12ecccb7.png)

**Comment by jdent02** [comment]
How do you have things set up now?  Holdouts become invisible in the alpha channel but the reflection and indirect lighting should still be visible in the rgb pass.  The idea is you combine the alpha of the beauty with a shadowpass from the appropriate filter and use that combined image for the compositing alpha

**Comment by SeanMorgan1962** [comment]
Hey, Jon.  Thanks for responding.  You've helped me in the past and I want you to know I appreciate the effort.  

Here are several screenshots of just a simple setup.  Just a plane, a cube, and one of the stock environment lights.  

Thanks, again.

Sean


![screenshot 21](https://user-images.githubusercontent.com/37384828/37439846-3d526830-27b7-11e8-89ce-2496fbe34edb.png)
![screenshot 20](https://user-images.githubusercontent.com/37384828/37439847-3d6dc418-27b7-11e8-8a39-a1f375706072.png)
![screenshot 23](https://user-images.githubusercontent.com/37384828/37439848-3d8a2324-27b7-11e8-9914-23c43a4f04bf.png)

**Comment by jdent02** [comment]
Okay that looks normal.  Try this:
Go to the scene tab and look for the sample filter section.  Add a ShadowFilter.  Pick some names for the occluded and unoccluded passes (doesn't really matter what) and put 'a' in the Shadow AOV box.  That will run the shadow filter and merge it into the alpha of your beauty pass.  That should give you the correct response for compositing.

Also be sure to turn off 'primary lamp visibility' for the environment map.  Otherwise it will show a solid alpha and that will screw things up.

![capture](https://user-images.githubusercontent.com/13521172/37441720-c2b57620-27d0-11e8-9cb6-e2bb2695ca90.PNG)
![scene 0001](https://user-images.githubusercontent.com/13521172/37441721-c2ce19fa-27d0-11e8-9d93-2928bdf3d0d2.png)

**Comment by SeanMorgan1962** [comment]
Jon, props to you, bud.  Worked like a charm.  

Completely off topic here but, I have been thinking of licensing Maya or 3DS Max but the cost has been prohibitive.  I thought Adobe products were pricey until I saw what Autodesk is asking.  Whew!  Are they that good?  Is it worth it?  Penny for your thoughts?  

Thanks, again for the pointers.

**Comment by jdent02** [comment]
Honestly I've never used either.  Some people love them, others not so much, and some people really hate them.

I do this stuff as a hobby so I can't justify shelling out money to Autodesk.  Heck one of the main reasons I started contributing to this plugin was out of gratitude to Pixar for not only making Renderman available to people like me, but also making it available to Blender.

Maya and 3DS Max are the industry standard though, so if you have the goal of being hired professionally you probably should learn them.  They do have an educational license that is really cheap but I think you have to provide proof that you're a student at an accredited institution.

On my own side note as awesome as it is to have Blender as an 'officially' supported DCC, I honestly don't know how long that will last with Brian gone.  If you want to stick with Renderman it might be better to do it with one of the tools that has more consistent attention from the developers.

**Comment by SeanMorgan1962** [comment]
Thanks for your insight, Jon.  

You make good points and I have to agree.  I'm 55, this is just a hobby for me, too.  I have a slim-to-none shot at being hired by any big production house, even working as a freelancer.  I still have a few before I am even experienced enough to ask someone to pay me for my work.  

Autodesk and Adobe both have a virtual stranglehold on the industry and they obviously know that.  SideFX and Foundry offer non-commercial versions of their apps.  I have Houdini and Nuke and they're fully-functional with minor limits on render output to keep people honest.  With Houdini, the learning is almost completely vertical it's so steep so, not being forced to pay to learn it is a blessing, itself. 

As far as RenderMan, much as I like their shaders, I think their days are numbered along with Autodesk and Adobe.  According to Ton Roosendaal, many larger VFX houses are looking at bringing Blender into their pipeline.  Why throw mountains of money at companies who offer sparse documentation, lousy support, and seemingly care only for customers who have the most money to spend?  

BTW, I noticed Brian is working with AMD on their ProRender renderer.  It seems worth looking at.  

Thanks again for the help.  

Sean

**Comment by jdent02** [comment]
I don't think Renderman is going anywhere.  Cycles is a great renderer for what it is, but it can't compete with Renderman (or Arnold) when it comes to the crazy complex stuff professional VFX houses are creating these days.  It doesn't have texture or geometry caches (yet) which are essential for the really big stuff.

As far as Blender, it is definitely making inroads (Pixar supposedly uses it for something, somewhere) but it's still got a long way to go before it could even so much as match parity with the big AD.  Maya and Max are entrenched and it's almost impossible to change that.  Plus they have a huge variety of high end plugins.

Don't get me wrong, I love Blender and don't much care for AD, but I really don't think Blender is ever going to dethrone Maya or Max in any meaningful way.  Most likely it'll end up in the same tier as Lightwave or Cinema 4D; used but under the radar.

**Comment by SeanMorgan1962** [comment]
When you put it that way, I have to concede the point.  I love RenderMan shaders...so realistic, from my limited experience.  Been eager to try Arnold, too.  Cycles is good and fast but I noticed as soon the poly count starts rising suddenly it's not so fast anymore.  And...don't dare add volumetrics into the mix.  (haha)  After what you said, I'm guessing that would be due to the lack caching for geo and materials?

As far as Blender, Mr.Roosendaal might've been blustering a bit and, now that you mentioned it, it occurs to me the amount of R&D necessary to bring Blender to the next level is going to take lots of money.  

Ultimately, I'd say if I want to play with the big boys, I will have to have big-boy toys.

**Comment by jdent02** [comment]
Yeah that's the truth of it.  Blender and Cycles is intentionally designed for small to mid size studios that don't have the financial or technical resources to stock up on all the pricey high end software (Maya, Nuke, Houdini, etc..).  From that angle it succeeds quite well, and Blender appears to be very popular in the video commercial industry (especially outside of North America).  But very few of those companies create imagery with the complexity of big FX houses like Weta or ILM.  That kind of stuff requires insane amounts of geometry and textures, combined with motion blur and depth of field.  From what I've heard, Cycles gets clobbered pretty hard in those areas.

Cycles isn't too bad at volumetrics now though.  Some recent patches have improved it's speed considerably, especially for smoke and fire.

But we're going way off topic here.  If you want to keep chatting shoot me a message on the Renderman forum.

**Comment by SeanMorgan1962** [comment]
No worries, Jon.  I will definitely take you up on the offer.  I always need to pick some brains with all this stuff.

**Comment by arabadzhiyski** [comment]
Jon, thanks for the shadow filter tip! I've been banging my head against the wall with that holdout workflow. Now, what else am I missing here... how do I get rid of the solid alpha of the holdout in the transmission areas (i.e. where glass objects overlap the holdout object) ? Here's a .gif that hopefully illustrates the issue better:

![holdout_alpha_issue](https://user-images.githubusercontent.com/41540889/43045913-6fe809e2-8dc1-11e8-96aa-b6f39ab71df1.gif)
